push/templates/push/validation.html

```diff
@@ -9,18 +9,22 @@
 {% endblock %}
 
 {% block content %}
-    <p>To verify you own the signing key for your app, please sign the following token using the same key you use to sign this application's VAPID JWT, and paste the url-safe, base64-encoded value below.</p>
-    <dl>
-        <dt>Token:</dt>
-        <dd>{{ app.vapid_key_token }}</dd>
-    </dl>
+    {% if app.vapid_key_status != "recording" %}
+        <p>To verify you own the signing key for your app, please sign the following token using the same key you use to sign this application's VAPID JWT, and paste the url-safe, base64-encoded value below.</p>
+        <dl>
+            <dt>Token:</dt>
+            <dd>{{ app.vapid_key_token }}</dd>
+        </dl>
 
-    <form id="vapid-validation" action="{% url 'push.validation' app.id %}" method="POST">
-        {% csrf_token %}
+        <form id="vapid-validation" action="{% url 'push.validation' app.id %}" method="POST">
+            {% csrf_token %}
 
-        <label for="{{ vapid_validation_form.signed_token.auto_id }}">URL-safe, base64-encoded signed token:</label>
-        {{ vapid_validation_form.signed_token }}
+            <label for="{{ vapid_validation_form.signed_token.auto_id }}">URL-safe, base64-encoded signed token:</label>
+            {{ vapid_validation_form.signed_token }}
 
-        <input type="submit" class="button" value="Validate">
-    </form>
+            <input type="submit" class="button" value="Validate">
+        </form>
+    {% else %}
+        <p>This application has already been validated.</p>
```

Comment:
Could say "This application was validated on {{ app.validated }}." to show the DateTime when the app was validated.